src/date.js

```diff
@@ -144,8 +144,8 @@ function formatDateBase(formatterString, date, utc) {
     o: () => calcISOWeek().year,
     Y: () => year,
     y: () => String(year).substr(2),
-    a: () => (hours <= 12 ? 'am' : 'pm'),
-    A: () => (hours <= 12 ? 'AM' : 'PM'),
+    a: () => (hours <= 12 ? hours == 12 ? 'pm' : 'am' : 'pm'),
+    A: () => (hours <= 12 ? hours == 12 ? 'PM' : 'AM' : 'PM'),

```

Comment:
The suggested Change is a bit convulted, this short version should do it, shouldn't it?

```suggestion
    a: () => (hours < 12 ? 'am' : 'pm'),
    A: () => (hours < 12 ? 'AM' : 'PM'),
```